Gordon360/Services/ProfileService.cs

```diff
@@ -426,6 +459,17 @@ public async Task UpdateImagePrivacyAsync(string username, string value)
         {
             MergeProfile(profile, JObject.FromObject(student));
             personType += "stu";
+
+            // Add graduation info if the student exists
+            if (student is StudentProfileViewModel studentProfile)
+            {
+                var graduationInfo = GetGraduationInfo(((StudentProfileViewModel)student).AD_Username);
+                if (graduationInfo != null)
+                {
+                    MergeProfile(profile, JObject.FromObject(graduationInfo));
+                }
+            }
+            
```

Comment:
Thank you for working on this, this is a cool improvement. However, I don't think this should be added to the ProfileViewModel. We are specifically trying to move away from having all a person's information bundled in one giant object, because it means that whenever we want one piece of information about that person, we have to fetch all the information about that person.

Having a separate API route (as you created above) just for this information, so that it's only fetched when it's relevant, is ideal.